### PR TITLE
DS-2: JDK8 Collections overview, covered by unit tests. #3 - Drafted …

### PR DESCRIPTION
**Pull request by vmalaya**
…implimentation of ArrayList unit testing

### CODE REVIEW

File: data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/AddingToArrayListTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        //given
+        ArrayList arrayList = new ArrayList();
+        String element = "str_new";
+        int index = 5;
+
+        //where
+
+        arrayList.add(index, element);
+
```
Comment (daggerok) [inline data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/AddingToArrayListTest.java]: remove useless line

File: data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/AddingToArrayListTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        String element = "str_new";
+        int index = 2;
+
+        //where
+        arrayList.add(index, element);
+
+        //then
+        Assert.assertTrue(arrayList.contains(element));
+    }
+
```
Comment (daggerok) [inline data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/AddingToArrayListTest.java]: remove useless lines

File: data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/AddingToArrayListTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        String element = "str_new";
+
+        // where
+       arrayList.add(element);
+
+        //then
+        Assert.assertTrue(arrayList.contains(element));
+    }
+
+    @Test (expected = IndexOutOfBoundsException.class)
```
Comment (daggerok) [inline data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/AddingToArrayListTest.java]: reformat the code

File: data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/InitializationTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        int size = arrayList.size();
+
+        // then
+        Assert.assertEquals(0,size);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void should__fail_init_array_list_by_given_capacity (){
+
+        //given
```
Comment (daggerok) [inline data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/InitializationTest.java]: please use same code style even for comments, I mean for example lines 24 and 31 use space everywhere or not use it but also everywhere

File: data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/InitializationTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        Collection collection = Arrays.asList(new Integer(1), new Integer(2), new Integer(3));
+        List arrayList;
+
+        // where
+        arrayList = new ArrayList(collection);
+
+        //then
+
```
Comment (daggerok) [inline data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/InitializationTest.java]: remove useless line

File: data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/InitializationTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        Assert.assertTrue(arrayList.containsAll(collection));
+
```
Comment (daggerok) [inline data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/InitializationTest.java]: remove useless line

File: data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/InitializationTest.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        // when
+        arrayList = new ArrayList(initCapacity);
+    }
+
+    @Test
+    public void should_create_array_list_from_given_collection (){
+
+        //given
+
```
Comment (daggerok) [inline data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/list/arraylist/InitializationTest.java]: remove useless line

File: data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/iterable/collection/set/HashSetTest.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.github.vmalaya.sigmasoftware.internship.datastructures.iterable.collection.set;
+
+/**
+ * Fast and efficient for adding and removing an element.
+ * You can use {@link java.util.HashSet} if you don’t
+ * want to maintain insertion order but want store unique objects.
+ */
+public class HashSetTest {
+
```
Comment (vmalaya) [inline data-structures/src/test/java/com/github/vmalaya/sigmasoftware/internship/datastructures/iterable/collection/set/HashSetTest.java]: keep same code style everywhere

**Pull request merged**